packages/util/src/environment.ts

```diff
@@ -68,13 +67,13 @@ export function isNode(): boolean {
  * Detect Browser Environment
  */
 export function isBrowser(): boolean {
-  return typeof window !== 'undefined';
+  return typeof self === 'object' && self.self === self;
```

Comment:
Just curious why checking self.self? Does some none browser env have self, but not self.self, or just to handle the case where someone defined a variable named self in none browse env? 

The intention of this function was just to check if it's a browser window env.
My intuition is that webworker probably needs a dedicated env check logic, but we don't need it anywhere. Do you see any implication of treating webworker as a browser env?

Reply:
To make sure that self is actually the global self, and not some random global that a developer defined.

I thought that in [the place you use this function](https://github.com/firebase/firebase-js-sdk/blob/f6afa45fe24f7adf5ea9a672100049c2ca8571db/packages/app/index.ts#L26), you'd want the result to be true for web workers as well. It's a useful check in both window and web worker contexts.